<h2>Syntax</h2>

<pre>
	<code>$argument.try($tryFunction, $catchFunction)</code>
</pre>

<h2>Description</h2>

<p>Executes a function and handles any errors that occur using a fallback function. The try function implements a try-catch behavior for TLScript that enables robust error handling.</p>

<p>The function works with three parameters:</p>

<ol>
	<li><strong>Argument</strong>: The value that is passed to the try function</li>
	<li><strong>Try function</strong>: The function to be executed that could cause an error</li>
	<li><strong>Catch function</strong>: The fallback function that is executed in the event of an error</li>
</ol>

<p>If the try function is executed successfully, its result is returned. If an error occurs, the catch function is executed with the error object as a parameter and its result is returned.</p>

<h2>Parameters</h2>

<table class="tlDocTable" style="height:155px; width:1517px">
	<thead>
		<tr>
			<th>Name</th>
			<th>Type</th>
			<th>Type Description</th>
			<th>Mandatory</th>
			<th>Standard</th>
		</tr>
		<tr>
			<td>argument</td>
			<td>Any</td>
			<td>The value that is passed to the try function as a parameter</td>
			<td>Yes</td>
			<td>-</td>
		</tr>
		<tr>
			<td>tryFunction</td>
			<td>Function</td>
			<td>The function that is to be executed. Receives the argument as a parameter</td>
			<td>yes</td>
			<td>-</td>
		</tr>
		<tr>
			<td>catchFunction</td>
			<td>function</td>
			<td>The fallback function that is executed when an error occurs. Receives the exception object as a parameter</td>
			<td>yes</td>
			<td>-</td>
		</tr>
	</thead>
</table>

<h2>Return value</h2>

<p><strong>Type:</strong> Object</p>

<p>Returns the result of the try function if it was executed successfully. If an error occurs, the result of the catch function is returned.</p>

<h2>Examples</h2>

<pre>
	<code>$emailAddress.try(
  emailAddr -&gt; sendMail(
    subject: $subject,
    to: $emailAddr,
    body: $emailBody
  ),
  catch: mailError -&gt; {
    info("E-Mail-Zustellung fehlgeschlagen: " + $mailError);
    false
  }
)</code>
</pre>

<p><strong>Output: </strong>Attempts to send an e-mail. If successful, the result of <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">sendMail</code> is returned. If an error occurs, <code class="bg-text-200/5 border border-0.5 border-border-300 text-danger-000 whitespace-pre-wrap rounded-[0.4rem] px-1 py-px text-[0.9rem]">false</code> and a user info are returned.</p>
